controllers/node/controllers/gpu_controller.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type GpuReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Logger logr.Logger
+}
+
+const (
+	GPU                                  = "gpu"
+	GPUInfo                              = "gpu-info"
+	GPUInfoNameSpace                     = "sealos"
+	NvidiaGPUProduct                     = "nvidia.com/gpu.product"
+	NvidiaGPUMemory                      = "nvidia.com/gpu.memory"
+	NvidiaGPU        corev1.ResourceName = "nvidia.com/gpu"
+	GPUProduct                           = "gpu.product"
+	GPUCount                             = "gpu.count"
+	GPUMemory                            = "gpu.memory"
+)
+
+//+kubebuilder:rbac:groups=node.k8s.io,resources=gpus,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=node.k8s.io,resources=gpus/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=node.k8s.io,resources=gpus/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=nodes,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=nodes/status,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
+
+func (r *GpuReconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
+	nodeList := &corev1.NodeList{}
```

Comment:
The node is not get updated frequently, so add cache+informer will almost prevent all calls to `api-server`.